lib/trino-record-decoder/src/main/java/io/trino/decoder/protobuf/ProtobufColumnDecoder.java

```diff
@@ -144,4 +177,36 @@ private static Optional<Descriptor> getDescriptor(FieldDescriptor fieldDescripto
         }
         return Optional.empty();
     }
+
+    private static Object createOneofJson(DynamicMessage message, Set<String> oneofColumns)
+    {
+        // Find the oneof field in the message; there will be at most one
```

Comment:
Is the "at most one" limitation of the protobuf?

Reply:
https://protobuf.dev/programming-guides/proto3/#using-oneof as far as I can tell, there could be more then just one named `oneof`s in the protobuf message.

Reply:
We should rather than pass the `oneofColumns` enumerate those using `DynamicMessage`

Reply:
You can specify more than one `oneof` fields in the message schema, but the field names across all `oneof` types have to be distinct so you'll only find one.

I've made the change to remove using `oneofColumns` in favor of enumerating the field names in `DynamicMessage`.

Reply:
> but the field names across all oneof types have to be distinct so you'll only find one.

Is this verified by the protobuf schema parser or something else? AFAIK the protobuf schema is based on tags so the schema itself probably doesn't care.

Reply:
Given a schema file:

```
syntax = "proto3";
message schema {
    oneof testOneofColumn {
        string stringColumn = 1;
    }
    string stringColumn = 2;
}
```

Loading the file-based schema parser doesn't result in any failure, but there is only one `stringColumn` field defined in the `Descriptor` (I'd say because it doesn't handle `oneof` types in the first place, so it is ignored).

The Confluent parser doesn't result in any failures, but there is only one `stringColumn` field in the resulting descriptor (it seems to flatten all of the `oneof` fields into a single field description, so having an additional field with the same name will be de-duplicated.

Meanwhile, `protoc` fails (specifically the Maven plugin which runs `protoc`) fails but this may just be an artifact of the code generation:

```
protoc-jar: executing: [/tmp/protocjar14095071959677501612/bin/protoc.exe, -I/tmp/protocjar6731110910677765846/include, -I/home/adamjshook/gitrepos/trino/lib/trino-record-decoder/src/test/resources/decoder/protobuf, --java_out=/home/adamjshook/gitrepos/trino/lib/trino-record-decoder/target/generated-test-sources, /home/adamjshook/gitrepos/trino/lib/trino-record-decoder/src/test/resources/decoder/protobuf/test_oneof.proto]
test_oneof.proto:7:12: "stringColumn" is already defined in "schema".
```

Reply:
Yeah, seems protobuf is under-specified for this case and behaviour depends on impl. Since messages will be produced using Confluent serializer i think the assumption holds true.

Reply:
Can we make it throw if we find multiple?